packages/ui/src/lib/button/Expandable.svelte

```diff
@@ -0,0 +1,46 @@
+<script lang="ts">
+import { faChevronDown, faChevronRight } from '@fortawesome/free-solid-svg-icons';
+import type { Snippet } from 'svelte';
+import Fa from 'svelte-fa';
+
+import { fadeSlide } from '../utils/animations';
+
+let {
+  expanded = $bindable(true),
+  initialized = $bindable(true),
+  title = undefined,
+  children = undefined,
+  onclick,
+}: {
+  expanded?: boolean;
+  initialized?: boolean;
+  title?: Snippet;
+  children?: Snippet;
+  onclick?: (expanded: boolean) => void;
+} = $props();
+
+function toggle(): void {
+  expanded = !expanded;
+  onclick?.(expanded);
+}
+</script>
+
+<div class="flex flex-col w-full">
+  <button onclick={(): void => toggle()} aria-expanded="{expanded}">
+    <div class="flex flex-row space-x-1 items-center">
+      {#if expanded}
+        <Fa class="w-4" icon={faChevronDown} />
+      {:else}
+        <Fa class="w-4" icon={faChevronRight} />
+      {/if}
+      {@render title?.()}
+    </div>
+  </button>
+{#if initialized}
+{#if expanded}
+  <div role="region" class="pt-5" transition:fadeSlide={{ duration: 250 }}>
```

Comment:
question: why do we need a new custom function vs

`import { fade, slide } from "svelte/transition";`
`transition:fade|slide`

Reply:
I hadn't touched the transition, I assume it was there originally because Svelte didn't support merged/composite transitions. Using this syntax doesn't work (linting says slide is unused, didn't work at runtime, can't find it in the docs) so I assume that's still the case in Svelte 5.

That said, the other way people get around the limitation is just nesting two transitions, so I did that instead - no more utility animation for same effect. 👍🏼

Reply:
thanks for the explanation